deepchecks/checks/performance/class_performance_imbalance.py

```diff
@@ -0,0 +1,239 @@
+                if dataset does not have a label column or the label dtype is unsupported;
+                if provided dataset is empty;
+        """
+        return self._class_performance_imbalance(dataset, model)
+
+    def _class_performance_imbalance(
+        self,
+        dataset: Dataset,
+        model: t.Any # TODO: find more precise type for model
+    ) -> CheckResult:
+        check_name = type(self).__name__
+        expected_model_types = [ModelType.BINARY, ModelType.MULTICLASS]
+
+        dataset.validate_label(check_name)
+        dataset.validate_features(check_name)
+        Dataset.validate_dataset(dataset, check_name)
+        task_type_validation(model, dataset, expected_model_types, check_name)
+
+        labels = t.cast(pd.Series, dataset.label_col())
+        features = t.cast(pd.DataFrame, dataset.features_columns())
+
+        if self.alternative_metrics is not None:
+            df = pd.DataFrame.from_dict(self._execute_alternative_metrics(
+                model, features, labels
+            ))
+        else:
+            y_true = labels
+            y_pred = model.predict(features)
+            unique_labels = get_unique_labels(labels, y_pred)
+            df = pd.DataFrame.from_dict({
+                name: dict(zip(unique_labels, metric_func(y_true, y_pred, labels=unique_labels))) # type: ignore
+                for name, metric_func in self._get_default_metrics().items()
+            })
+
+        def display():
+            title = (
+                'Class Performance Imbalance Check for binary data'
+                if len(labels) == 2
+                else 'Class Performance Imbalance Check for multi-class data'
+            )
+
+            df.transpose().plot.bar(
+                title=title,
+                backend='matplotlib',
+                xlabel='Metrics',
+                ylabel='Values'
+            )
+
+        return CheckResult(
+            value=df.transpose().to_dict(),
+            header='Class Performance Imbalance',
+            check=type(self),
+            display=display
+        )
+
+    def _get_default_metrics(self) -> t.Dict[str, t.Callable[..., np.ndarray]]:
+        return {
+            'Accuracy': partial(recall_score, zero_division=0, average=None),
+            'Precision': partial(precision_score, zero_division=0, average=None),
+            'Recall': partial(recall_score, zero_division=0, average=None)
+        }
+
+    def _execute_alternative_metrics(
+        self,
+        model: t.Any,
+        features: pd.DataFrame,
+        labels: pd.Series,
+    ) -> t.Dict[str, t.Dict[t.Hashable, t.Union[int, float]]]:
+        check_name = type(self).__name__
+        metrics = t.cast(t.Mapping[str, MetricFunc], self.alternative_metrics)
+        result: t.Dict[str, t.Dict[t.Hashable, t.Union[int, float]]] = {}
+
+        for metric_name, metric_func in metrics.items():
+            metric_result = metric_func(model, features, labels)
```

Comment:
I still don't understand how metric function returns a dict

Reply:
from the `precision_score` doc string:

```
average : {'micro', 'macro', 'samples', 'weighted', 'binary'} or None, default='binary'
        This parameter is required for multiclass/multilabel targets.
        If ``None``, the scores for each class are returned. Otherwise, this
        determines the type of averaging performed on the data:
```

Reply:
Mehdish14